content/blog/2021-09-24-refactor.md

```diff
@@ -0,0 +1,155 @@
+﻿---
+title: Easy Stuctural Refactors to Python Source Code
+date: 2021-09-24
+description: |
+  Simple, hassle-free, dependency-free, AST based source code refactoring toolkit.
+
+descriptionLong: |
+  [Refactor](https://github.com/isidentical/refactor) is a source code 
+  refactoring engine. By taking advantage of the Python standard library's
+  [AST](https://docs.python.org/3/library/ast.html) module, we can find-and-fix
+  patterns in massive codebases.
+
+picture: 2021-09-24/refactor.png
+author: batuhan_taskaya
+commentsUrl: https://discuss.dvc.org/t/easy-structural-refactors-to-python-source-code/895
+tags:
+  - Refactor
+  - AST
+  - Open-source
+  - Engineering
```

Comment:
```suggestion
  - Refactor
  - Release
  - Python
  - AST
  - Open-source
  - Engineering
```

Reply:
I'm good with this.  @jorgeorpinel @shcheklein ? I know there was some concern about having too many tags.  Python for sure is good not sure, about "Release"

Reply:
this is based on past tags. Some of the others in fact are dubious (have complained about this before - will open follow-up issue)

Reply:
already here: https://github.com/iterative/dvc.org/issues/1657#issuecomment-668778880

Reply:
<details>

num|Tag
--:|:--
26|Community
22|CML
21|Heartbeat
20|MLOps
13|CI/CD
10|Pipelines
10|DVC
 9|Meetup
 6|Release
 6|R
 6|Python
 6|Plots
 6|DataOps
 5|Tutorial
 4|Reproducibility
 4|Google Season of Docs
 4|Git
 3|Hacktoberfest
 3|Experiments
 3|DevOps
 3|Best Practices
 2|Vega
 2|Terraform
 2|SciPy
 2|Remote Storage
 2|PyCon
 2|Open Source Summit
 2|Mentoring
 2|Import
 2|Hyperparameters
 2|Google Drive
 2|GitHub Actions
 2|DVC 1.0
 2|Documentation
 2|Docker
 2|DivOps
 2|Conference
 2|Cache
 2|Ambassador
 1|YouTube
 1|Volunteer
 1|Videos
 1|Udemy Course
 1|Udemy
 1|Tags
 1|Tab
 1|Students
 1|SSH
 1|shtab
 1|Self-hosted Runner
 1|Runners
 1|RStats
 1|Remotes
 1|Reddit
 1|Rclone
 1|PyTorch
 1|PyData
 1|PTDC-18
 1|Project
 1|Productivity
 1|Podcast
 1|Performance
 1|Optimization
 1|New feature
 1|Monorepo
 1|Model Ensembling
 1|MinIO
 1|Metrics
 1|Machine Learning
 1|Homebrew
 1|GPUs
 1|Google Summer of Code
 1|Google Cloud Storage
 1|Git LFS
 1|GitLab CI
 1|GCP
 1|External Data
 1|Engineering
 1|Discord
 1|Data registry
 1|Data
 1|DAGsHub
 1|Conda
 1|Completion
 1|Cloud
 1|CLI
 1|Book
 1|Blogging
 1|Bitbucket
 1|Benchmark
 1|Azure
 1|Autocomplete

</details>

Reply:
Is there a way we can check how many times people have actually clicked on one of the tags when they get to the bottom of the article to look at related content? @julieg18 @rogermparent   I'm guessing it's minute.  The search ability at the top of the blog will be more useful. Yay Q4 plan!

Regarding Release tag, release has referred to our releases not outside releases, I think, but if the tags are used it will send people to our release info, which is desirable.

Reply:
I'm good with the tags.  Can we please merge this and not put it off 'til next week? @shcheklein @jorgeorpinel

Reply:
> Is there a way we can check how many times people have actually clicked on one of the tags when they get to the bottom of the article to look at related content?

I'd imagine we could hook up an analytics event to those tags to be able to collect this info, I haven't done much analytics configuration but it's pretty standard stuff.

Reply:
Tags should be also propagated into keywords? (SEO).

Also, I think we'll grow tags like the `engineering` one and it can become a sub-blog that will be important to have for eng team.